tests/test_hdf_writer.py

```diff
@@ -0,0 +1,82 @@
+import h5py
+from geometry_constructor.Writers import HdfWriter
+from geometry_constructor.Models import InstrumentModel, Detector, PixelGrid, Corner, CountDirection
+
+
+def assess_pixel_grid_direction(direction, corner, expected_data):
```

Comment:
See if you can use the "in-memory" file option to avoid having to do disk IO in tests.
http://assorted-experience.blogspot.com/2015/06/h5py-in-memory-files.html

Reply:
Looks like to completely avoid disk writes we'd have to avoid closing the file in the save function, or it'd be dropped from memory. I don't think we should be leaving the file open, so don't think we should use the purely in memory option.

It looks like the hybrid option will still write to file, but keep a copy in memory which may reduce the number of disk ops. However then we'd be testing the code with a mode that it wouldn't be used in for the real app, unless you want the memory mapped mode to be used in production too?